app/controllers/application_controller.rb

```diff
@@ -5,6 +5,152 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect "/tweets"
+    end
+    erb :'/users/create_user'
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect "/signup"
+    else
+
+      @user = User.create(params)
+      session[:user_id] = @user.id
+      redirect "/tweets"
+    end
+
+    erb :'/users/create_user'
+  end
+
+  get '/login' do
+    if logged_in?
+      redirect "/tweets"
+    end
+    erb :'/users/login'
+  end
+
+  post '/login' do
+    @user = User.find_by_username(params[:username])
+
+    if User.all.include?(@user) && @user.authenticate(params[:password])
+      session[:user_id] = @user.id
+
+      redirect "/tweets"
+    end
+
+    erb :'/users/login'
+  end
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+      redirect "/login"
+    end
+    redirect "/tweets"
+  end
+
+  get '/tweets' do
+    @tweets = Tweet.all
+
+    if !logged_in?
+      redirect "/login"
+    end
+    erb :'/tweets/tweets'
+  end
+
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+
+    erb :'/users/show'
+  end
+
+  get '/tweets/new' do
+    if !logged_in?
+      redirect "/login"
+    end
+
+    erb :'/tweets/create_tweet'
+  end
+
+  post '/tweets' do
+    if params[:content] == ""
+      redirect "/tweets/new"
+    elsif logged_in?
+      @tweet = Tweet.create(content: params[:content], user_id: session[:user_id])
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.